### ISSUE TITLE
Add patch file and install instruction for the CentOS7 x86_64 - [closed]

### ISSUE THREAD

**Issue by DavidGriffith** [DavidGriffith/minipro-import-test#309]
In GitLab by @YouSysAdmin on Feb 26, 2015, 03:44

_Merges SysAlex:CentOS7_Patch -> master_

Патч и инструкция для CentOS7 x86_64

**Comment by DavidGriffith** [comment]
In GitLab by @YouSysAdmin on Jul 10, 2015, 04:06

**Версия для RedHat систем** 
**RedHat version** 
https://github.com/POStroi/minipro_centos7

**Comment by DavidGriffith** [comment]
It seems that the impetus of this patch is a hangup between libusb_strerror() and libusb_error_name().

**Comment by DavidGriffith** [comment]
*Created by: chenz*

minipro uses libusb 1, while Centos 7 only has libusbx, which is an enhanced fork of libusb 0 that was obsoleted once libusb 1 came out.

Also, the udev configuration is adapted - although I'm not sure if it is out of necessity or preference (device is created world-writeable).

**Comment by DavidGriffith** [comment]
Why does minipro need to do anything special to support libusbx given the latter is a drop-in for libusb 1.x and given that libusbx merged with libusb 1.x around four years ago?

**Comment by DavidGriffith** [comment]
*Created by: chenz*

Actually, the code changes should already be obsolete as of 4b49d04aa8752784daf8ff9e2ac36f370afd44b7.

Not sure about the build instructions, would `pkg-config --cflags libusb-1.0` work with libusbx? (EDIT: the point is, the instructions would have to be adjusted to include overriding libusb_CFLAGS etc).

**Comment by DavidGriffith** [comment]
Does anyone reading this have a CentOS 7 installation and know it well enough to investigate this problem?  I'd like to avoid having to do anything special to add support for libusbx and instead encourage installation of libusb1.

**Comment by DavidGriffith** [comment]
*Created by: chenz*

I'll investigate this later today in virtualization. As I mentioned, libusbx should no longer be an issue here except for documentation. The udev stuff is more interesting to me at this point - I'll look at related packages in Centos like I did for Debian. Then maybe we can come up with either a unified solution, or proper distribution-specific distinctions.

**Comment by DavidGriffith** [comment]
It would also be nice to get in touch with some diehards of Redhat/Fedora/CentOS, of Suse, and whoever else seems like a good idea so we can get packaging stuff done for those distributions.

**Comment by DavidGriffith** [comment]
*Created by: chenz*

I have checked out Centos 7. The code compiles without modification with `libusbx-devel` installed (ie `pkg-config libusb-1.0 ...` works).

The udev rules are another matter. As I already mentioned, the rules file from the patch makes the device world-writable. The group `plugdev`, which the debian-like rules file uses does not exist on Centos. I have not yet found a comparable package to see how it handles udev (I found an avrdude package, but it did not contain any rules). But clearly, we can already see the rules are distro-specific. 

I propose the following:
- Rename the existing udev rules file to `udev/debian/60-minipro.rules`. This is the right thing to do, because we now declare the file "debianlike"-specific (which it implicitly already was), and 80 is clearly the wrong priority on debianlike systems - it must be < 71, as explained in #68.
- Add the rules file from the this patch as-is as `udev/centos7/80-minipro.rules`. It does not contain the uaccess rule so the 71 priority limit is not an issue. If someone finds a problem with the file: good, we can fix it then (now it is just hidden in a patch in a PR).
- Update documentation accordingly
- (Also dont forget to adjust the deb build to the new file location)

EDIT: let me know what you think, I can then prepare a PR
EDIT2: Fixed file names
